New CSS Architecture

gemfarmer opened the pull request:
@shawnbot @meiqimichelle @ericronne I'd like to get your eyes on this.

This address #637 (fine tuning the glossary appearance) and #602 (new css architecture)

What does it do?
It creates a new _sass directory structure with the following directories:
* components – single elements or small clusters of elements that contain self-contained styling
* elements – these are eiti specific styles on commonly used elements such as links, lists, forms, typography
* extends – 
* layouts – styles for site structure
* lib – third party styles
* mixins – re-usable functions that are used to define styles site wide
* modules – Complex pieces made of components and elements

** Note: There are several 'modules' that I haven't broken apart or cleaned up yet

meiqimichelle commented:
Besides @shawnbot's note, this looks good to me! I ran it locally, and spot checked the file system -- :+1:

gemfarmer commented on .gitignore:
```diff
@@ -11,3 +11,4 @@ bower.json
 bower_components
 node_modules
 npm-debug.log
+styleguide
```
Ah, you are right! It will be included in the next push.

gemfarmer commented:
Since friday I...
* added all the changes made on #644
* removed the `_scss/modules` directory and moved all the scss files in it to `_scss/components`
* added a `_scss/blocks` directory to house all the non-component page specific code
* changed import structure components, elements, and blocks to import from `all`.
